### ISSUE TITLE
Multi GPU Understanding

### ISSUE THREAD

**Issue by storm221221** [AlexeyAB/darknet#6492]
I have a couple of questions regarding multi GPU training 

1-> I've train first 1000 iteration with single GPU which tack around 2:30 h after that I used that weight and started training in 2 GPU but 1000-2000 iteration also takes around 2:30 h. It should be half the time right?

2-> I've set network height/width size to 608 and batch and subdivision to 64(due to memory issues) but after 1000 iteration when I use 2 GPU I'm not able to decrease subdivision due to Cuda memory error so, my question is in multi GPU training how minibatch is provided to each GPU. Is it loading two different batches in each GPU and uses a minibatch? If yes then timing should be half. but I'm getting same timing in single GPU training and multi GPU training

I'm using NVIDIA 1080ti

**Comment by AlexeyAB** [comment]
1. May be performance bottleneck on your CPU / HDD, or you are doing something wrong
2. It doesn't use batch-sync, so you can't increase mini-batch by using 2 x GPUs. Use `[net] mosaic=1` instead.

Do you use 2 x NVIDIA 1080ti ? What command do you use?

**Comment by storm221221** [comment]
Thanks for the answer.
1. Yes, I'm using 2 1080ti.
2. Command I used : "./darknet detector train ./obj.data ./yolov4_coustom.cfg ./yolov4_1000.weights -gpus 0,1"

**Comment by storm221221** [comment]
I didn't get your point on 2nd answer mosaic is for data augmentation how it'll help me in batch? 
-> Can you also explain in multi GPU training how batch will get devided. For example if we use single GPU then minibatch is applied to GPU where minibatch = batch/subdivision. So, in case of multi GPU training batch are separately loaded to each GPU(1st GPU will get full batch and 2nd GPU will also get full batch)?

**Comment by AlexeyAB** [comment]
mosaic=1 reduces the need for a large minibatch: https://arxiv.org/pdf/2004.10934.pdf

> Mosaic represents a new data augmentation method that mixes 4 training images. Thus 4 different contexts are mixed, while CutMix mixes only 2 input images. This allows detection of objects outside their normal context. In addition, batch normalization calculates activation statistics from 4 different images on each layer. **This significantly reduces the need for a large mini-batch size.**

If you use 2 x GPUs, then mini-batch will be the same, just batch will be 2x larger.

Show such screenshot during training
![image](https://user-images.githubusercontent.com/4096485/90321016-b37d5500-df4e-11ea-84a6-c6dd58403d0d.png)
